test/functional/test_framework/wallet_cli_controller.py

```diff
@@ -207,12 +215,16 @@ async def new_address(self) -> str:
     async def list_utxos(self, utxo_types: str = '', with_locked: str = '', utxo_states: List[str] = []) -> List[UtxoOutpoint]:
         output = await self._write_command(f"account-utxos {utxo_types} {with_locked} {''.join(utxo_states)}\n")
 
-        pattern = r'UtxoOutPoint\s*{[^}]*Id<Transaction>\{([^}]*)\}[^}]*index:\s*(\d+)'
-        matches = re.findall(pattern, output, re.DOTALL)
-        return [UtxoOutpoint(id=match[0].strip(), index=int(match[1].strip())) for match in matches]
+        j = json.loads(output)
 
-    async def get_transaction(self, tx_id: str) -> str:
-        return await self._write_command(f"transaction-get {tx_id}\n")
+        return [UtxoOutpoint(id=match["outpoint"]["id"]["Transaction"], index=int(match["outpoint"]["index"])) for match in j]
+
+    async def get_transaction(self, tx_id: str):
+        out = await self._write_command(f"transaction-get {tx_id}\n")
+        try:
+            return json.loads(out)
+        except:
+            return out
```

Comment:
Why is this necessary? Shouldn't it be either one or the other?

Reply:
tries to load it as a json, but error messages are not in json format, so it fails and we return the string as is for matching in the test